http/http-api/src/main/java/com/okta/commons/http/QueryString.java

```diff
@@ -35,18 +43,29 @@ public QueryString(){}
 
     public QueryString(Map<String,?> source) {
         super();
+
         if (!Collections.isEmpty(source)) {
-            for(Map.Entry<String,?> entry : source.entrySet()) {
+            for (Map.Entry<String,?> entry : source.entrySet()) {
                 String key = entry.getKey();
                 Object value = entry.getValue();
-                String sValue = value != null ? String.valueOf(value) : null;
+                String sValue = null;
+                if (value != null) {
+                    if (value instanceof Date) {
+                        sValue = RequestUtils.getFormattedDate(String.valueOf(value));
```

Comment:
Why is there a `String.valueOf(value)` here? Keep the value a date until it is parsed
```suggestion
                        sValue = RequestUtils.getFormattedDate((Date) value);
```